src/rgw/rgw_admin.cc

```diff
@@ -5213,12 +5213,6 @@ int main(int argc, const char **argv)
   }
 
   if (opt_cmd == OPT_OLH_GET) {
-    RGWBucketInfo bucket_info;
-    int ret = init_bucket(tenant, bucket_name, bucket_id, bucket_info, bucket);
-    if (ret < 0) {
-      cerr << "ERROR: could not init bucket: " << cpp_strerror(-ret) << std::endl;
-      return -ret;
-    }
```

Comment:
see that these calls to `init_bucket()` are duplicated above in the `if (opt_cmd == OPT_OLH_GET || opt_cmd == OPT_OLH_READLOG) {` block

but that block initializes a locally-scoped `RGWBucketInfo bucket_info;`, rather than the one in `main()`'s scope. so the call to `store->get_olh()` here would be using an uninitialized `bucket_info`

so it looks like you also need to remove that extra `RGWBucketInfo bucket_info;` in the block above